euphonic/powder.py

```diff
@@ -180,18 +193,24 @@ def sample_sphere_structure_factor(
             physical_property='coherent_scattering_length',
             collection=scattering_lengths)  # type: dict
 
-    if dw is None:
-        dw_qpts = mp_grid([20, 20, 20])
-        dw_phonons = fc.calculate_qpoint_phonon_modes(dw_qpts)
-        dw = dw_phonons.calculate_debye_waller(temperature
-                                               )  # type: DebyeWaller
+    if temperature is not None:
+        if (dw is None):
+            dw_qpts = mp_grid([20, 20, 20])
```

Comment:
This could definitely benefit from the 1 parameter improved MP sampling...

Reply:
Done!